ql/src/java/org/apache/hadoop/hive/ql/plan/PlanUtils.java

```diff
@@ -1244,16 +1247,31 @@ static Map getPropertiesExplain(Properties properties) {
         }
         clone.remove(StatsSetupConst.NUM_ERASURE_CODED_FILES);
       }
-      if (properties.containsKey(TABLE_IS_CTAS)) {
-        if (clone == null) {
-          clone = new HashMap<>(properties);
+      for (String key : FILTER_OUT_FROM_EXPLAIN) {
```

Comment:
would it work for the regular CTAS without the explain?

Reply:
yes, it is only necessary for Iceberg